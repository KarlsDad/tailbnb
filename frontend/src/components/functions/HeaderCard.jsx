import * as React from 'react';
import { useState, useRef, useEffect } from 'react';
import { TweenLite, Power3 } from 'gsap';
import {
  calculateGuests,
  calculateInfants
} from '../../util/GuestModalFunctions';
import { rotateArrow, backRotateArrow } from '../../animations/arrow';

// Modals
import { CalendarModal } from '../modals/CalendarModal';
import { GuestPickerModal } from '../modals/GuestPickerModal';

export const HeaderCard = () => {
  // Check in
  const [checkInModal, setCheckInModal] = useState(false);
  const [checkInDate, setCheckInDate] = useState(['dd', 'mm', 'yy']);

  // Check out
  const [checkOutModal, setCheckOutModal] = useState(false);
  const [checkOutDate, setCheckOutDate] = useState(['dd', 'mm', 'yy']);

  // Guest
  const [guestModal, setGuestModal] = useState(false);
  const [adultNumber, setAdultNumber] = useState(0);
  const [childrenNumber, setChildrenNumber] = useState(0);
  const [infantNumber, setInfantNumber] = useState(0);

  let guestArrow1 = useRef(null);
  let guestArrow2 = useRef(null);

  return (
    <div className='sm:w-full md:w-100 md:pt-8 md:pb-6 md:mt-10 bg-white md:rounded md:shadow-2xl py-4'>
      <div className='mx-6 md:mx-8'>
        <h3
          style={{ fontFamily: 'airbnb-bold' }}
          className='hidden md:inline-block md:text-3xl text-gray-750 leading-tight'>
          Book unique places to <br />
          stay and things to do.
        </h3>
        <form>
          <div className='relative'>
            <label
              style={{ fontFamily: 'airbnb-bold' }}
              className='block mt-5 mb-2 text-xs text-gray-750 uppercase tracking-wider'
              htmlFor='where'>
              Where
            </label>
            <input
              style={{ fontFamily: 'airbnb-book' }}
              className='w-full border border-gray-300 p-2 rounded placeholder-gray-900 tracking-wide'
              type='text'
              id='where'
              placeholder='Anywhere'
            />
          </div>
          <div className='flex flex-wrap items-stretch justify-start relative'>
            <div className='w-1/2'>
              <label
                style={{ fontFamily: 'airbnb-bold' }}
                className='block mt-5 mb-2 text-xs text-gray-750 uppercase'
                htmlFor='checkin'>
                Check-In
              </label>
              <div className='relative w-full'>
                <div className='w-full flex justify-start'>
                  {checkInModal ? (
                    <button
                      onClick={(e) => {
                        e.preventDefault();
                        setCheckInModal(!checkInModal);
                      }}
                      style={{ fontFamily: 'airbnb-book' }}
                      className='w-full border border-green-850 py-3 rounded rounded-l-none placeholder-gray-900 tracking-wide'>
                      dd-mm-yy
                    </button>
                  ) : (
                    <button
                      onClick={(e) => {
                        e.preventDefault();
                        setCheckInModal(!checkInModal);
                      }}
                      style={{ fontFamily: 'airbnb-book' }}
                      className='pr-16 w-full border border-gray-300 py-3 rounded rounded-l-none placeholder-gray-900 tracking-wide'>
                      {`${checkInDate[0]}/${checkInDate[1]}/${checkInDate[2]}`}
                    </button>
                  )}
                </div>
                {checkInModal ? (
                  <CalendarModal setCheckDate={setCheckInDate} />
                ) : null}
              </div>
            </div>
            <div className='w-1/2'>
              <label
                style={{ fontFamily: 'airbnb-bold' }}
                className='block mt-5 mb-2 text-xs uppercase text-gray-750'
                htmlFor='checkout'>
                Check-Out
              </label>
              <div className='relative w-full'>
                <div className='w-full flex justify-start'>
                  {checkOutModal && !checkInModal ? (
                    <button
                      onClick={(e) => {
                        e.preventDefault();
                        setCheckOutModal(!checkOutModal);
                      }}
                      style={{ fontFamily: 'airbnb-book' }}
                      className='w-full border border-green-850 py-3 rounded rounded-l-none placeholder-gray-900 tracking-wide'>
                      dd-mm-yy
                    </button>
                  ) : (
                    <button
                      onClick={(e) => {
                        e.preventDefault();
                        setCheckOutModal(!checkOutModal);
                      }}
                      style={{ fontFamily: 'airbnb-book' }}
                      className='pr-16 w-full border border-gray-300 py-3 rounded rounded-l-none placeholder-gray-900 tracking-wide'>
                      {`${checkOutDate[0]}/${checkOutDate[1]}/${checkOutDate[2]}`}
                    </button>
                  )}
                </div>
                {checkOutModal && !checkInModal ? (
                  <div
                    style={{ top: -50, right: 115 }}
                    className='bg-red-550 relative'>
                    <CalendarModal setCheckDate={setCheckOutDate} />
                  </div>
                ) : null}
              </div>
            </div>
          </div>
          <div className='relative'>
            <label
              style={{ fontFamily: 'airbnb-bold' }}
              className='block mt-5 mb-2 text-xs uppercase text-gray-750'
              htmlFor='guests'>
              Guests
            </label>
            <div className='inline-block relative w-full'>
              {guestModal ? (
                <button
                  style={{ fontFamily: 'airbnb-book' }}
                  onClick={(e) => {
                    e.preventDefault();
                    rotateArrow(guestArrow1);
                    setGuestModal(!guestModal);
                  }}
                  className='appearance-none pl-2 w-full border border-green-850 py-3 text-left rounded placeholder-gray-900 tracking-wide'>
                  {calculateGuests(adultNumber, childrenNumber)}
                  {adultNumber > 0 && infantNumber > 0
                    ? `, ${calculateInfant(adultNumber, infantNumber)}`
                    : null}
                </button>
              ) : (
                <button
                  style={{ fontFamily: 'airbnb-book' }}
                  onClick={(e) => {
                    e.preventDefault();
                    backRotateArrow(guestArrow2);
                    setGuestModal(!guestModal);
                  }}
                  className='appearance-none pl-2 w-full border border-gray-300 py-3 text-left rounded placeholder-gray-900 tracking-wide'>
                  {calculateGuests(adultNumber, childrenNumber)}
                  {adultNumber > 0 && infantNumber > 0
                    ? `, ${calculateInfant(adultNumber, infantNumber)}`
                    : null}
                </button>
              )}
              {guestModal ? (
                <div
                  ref={(el) => (guestArrow1 = el)}
                  style={{}}
                  className='pointer-events-none flex absolute inset-y-0 right-0 items-center px-2 text-gray-700 text-black'>
                  <svg
                    style={{ fill: '#222222' }}
                    className='h-4 w-4'
                    xmlns='http://www.w3.org/2000/svg'
                    viewBox='0 0 20 20'>
                    <path d='M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z' />
                  </svg>
                </div>
              ) : (
                <div
                  ref={(el) => (guestArrow2 = el)}
                  style={{}}
                  className='pointer-events-none flex absolute inset-y-0 right-0 items-center px-2 text-gray-700 text-black'>
                  <svg
                    className='h-4 w-4'
                    xmlns='http://www.w3.org/2000/svg'
                    viewBox='0 0 20 20'>
                    <path d='M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z' />
                  </svg>
                </div>
              )}
            </div>
            {guestModal ? (
              <GuestPickerModal
                incrementAdult={() => setAdultNumber(adultNumber + 1)}
                decrementAdult={() => setAdultNumber(adultNumber - 1)}
                incrementChildren={() => setChildrenNumber(childrenNumber + 1)}
                decrementChildren={() => setChildrenNumber(childrenNumber - 1)}
                incrementInfant={() => setInfantNumber(infantNumber + 1)}
                decrementInfant={() => setInfantNumber(infantNumber - 1)}
                adult={adultNumber}
                children={childrenNumber}
                infant={infantNumber}
                resetNumber={() => {
                  setAdultNumber(0);
                  setChildrenNumber(0);
                  setInfantNumber(0);
                }}
                switchModal={() => setGuestModal(false)}
              />
            ) : null}
          </div>
        </form>
        <div className='w-full flex items-center md:justify-end md:flex-wrap mt-6 rounded-lg'>
          <button className='w-full md:w-auto md:inline-block inline-block py-3 px-6 bg-red-550 rounded text-white font-bold'>
            Search
          </button>
        </div>
      </div>
    </div>
  );
};
