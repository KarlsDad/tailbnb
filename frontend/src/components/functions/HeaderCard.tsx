import * as React from 'react';

export const HeaderCard: React.FC<{}> = () => {
  return (
    <div className='sm:w-full md:w-100 md:pt-8 md:pb-6 md:mt-10 bg-white md:rounded md:shadow-2xl py-4'>
      <div className='mx-4 md:mx-8'>
        <h3
          style={{ fontFamily: 'airbnb-bold' }}
          className='hidden md:inline-block md:text-3xl text-gray-750 leading-tight'>
          Book unique places to <br />
          stay and things to do.
        </h3>
        <form>
          <div className=''>
            <label
              style={{ fontFamily: 'airbnb-bold' }}
              className='block mt-5 mb-2 text-xs text-gray-750 uppercase tracking-wider'
              htmlFor='where'>
              Where
            </label>
            <input
              style={{ fontFamily: 'airbnb-book' }}
              className='w-full border border-gray-300 p-2 rounded placeholder-gray-900 tracking-wide'
              type='text'
              id='where'
              placeholder='Anywhere'
            />
          </div>
          <div className='flex flex-wrap items-stretch justify-start'>
            <div className='w-1/2'>
              <label
                style={{ fontFamily: 'airbnb-bold' }}
                className='block mt-5 mb-2 text-xs text-gray-750 uppercase'
                htmlFor='checkin'>
                Check-In
              </label>
              <input
                style={{ fontFamily: 'airbnb-book' }}
                className='pl-2 w-full border border-gray-300 py-3 rounded rounded-r-none placeholder-gray-900 tracking-wide'
                type='text'
                id='checkin'
                placeholder='dd-mm-yyyy'
              />
            </div>
            <div className='w-1/2'>
              <label
                style={{ fontFamily: 'airbnb-bold' }}
                className='block mt-5 mb-2 text-xs uppercase text-gray-750'
                htmlFor='checkout'>
                Check-Out
              </label>
              <input
                style={{ fontFamily: 'airbnb-book' }}
                className='pl-2 w-full border border-gray-300 py-3 rounded rounded-l-none placeholder-gray-900 tracking-wide'
                type='text'
                id='checkout'
                placeholder='dd-mm-yyyy'
              />
            </div>
          </div>
          <div>
            <label
              style={{ fontFamily: 'airbnb-bold' }}
              className='block mt-5 mb-2 text-xs uppercase text-gray-750'
              htmlFor='guests'>
              Guests
            </label>
            <div className='inline-block relative w-full'>
              <select
                style={{ fontFamily: 'airbnb-book' }}
                className='appearance-none pl-2 w-full border border-gray-300 py-3 rounded rounded-l-none placeholder-gray-900 tracking-wide'
                id='guests'>
                <option value='1'>2 Guests</option>
                <option value='2'>3 Guests</option>
                <option value='3'>4 Guests</option>
              </select>
              <div className='pointer-events-none flex absolute inset-y-0 right-0 items-center px-2 text-gray-700 text-black'>
                <svg
                  className='fill-current h-4 w-4'
                  xmlns='http://www.w3.org/2000/svg'
                  viewBox='0 0 20 20'>
                  <path d='M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z' />
                </svg>
              </div>
            </div>
          </div>
        </form>
        <div className='w-full flex items-center md:justify-end md:flex-wrap mt-6 rounded-lg'>
          <button className='w-full md:w-auto md:inline-block inline-block py-3 px-6 bg-red-550 rounded text-white font-bold'>
            Search
          </button>
        </div>
      </div>
    </div>
  );
};
