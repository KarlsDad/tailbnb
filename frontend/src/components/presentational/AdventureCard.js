import React, { useState } from 'react';
import LazyImage from 'react-lazy-progressive-image';
import Link from 'next/link';

export default ({ id, img, imglow, country, title, cost, period }) => {
  return (
    <Link href='/adventures/[id]' as={`/adventures/${id}`}>
      <div className='w-30/31 cursor-pointer'>
        <LazyImage src={img} placeholder={imglow}>
          {(src, loading, isVisible) => (
            <img className='w-full rounded-lg' src={src} />
          )}
        </LazyImage>

        <p className='uppercase text-xs my-2 font-semibold'>{country}</p>
        <p className='my-1 w-5/6'>{title}</p>
        <p>
          From ${cost} / person
          <span> Â· </span>
          {period} days
        </p>
      </div>
    </Link>
  );
};
