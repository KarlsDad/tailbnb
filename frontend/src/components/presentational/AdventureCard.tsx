import * as React from 'react';
import Link from 'next/link';

interface Prop {
  id: string;
  img: string;
  country: string;
  title: string;
  cost: number;
  period: number;
}

export const AdventureCard: React.FC<Prop> = ({
  id,
  img,
  country,
  title,
  cost,
  period
}) => {
  return (
    <Link href='/adventures/[id]' as={`/adventures/${id}`}>
      <div className='w-30/31 cursor-pointer'>
        <img className='w-full rounded-lg' src={img} />

        <p className='uppercase text-xs my-2 font-semibold'>{country}</p>
        <p className='my-1 w-5/6'>{title}</p>
        <p>
          From ${cost} / person
          <span> Â· </span>
          {period} days
        </p>
      </div>
    </Link>
  );
};
