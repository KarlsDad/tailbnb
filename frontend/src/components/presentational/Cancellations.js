import React from 'react';

export default () => {
  return (
    <div className='border-b border-gray-300 mt-4'>
      <div className='border-gray-300'>
        <h3 style={{ fontFamily: 'airbnb-bold' }} className='text-gray-750'>
          Cancellations
        </h3>
        <div className=''>
          <p
            style={{ fontFamily: 'airbnb-medium' }}
            className='mt-4 text-gray-750'>
            Free cancellation for 48 hours
          </p>
          <p style={{ fontFamily: 'airbnb-book' }} className='text-gray-750'>
            After that, cancel up to 7 days before check-in and get a 50%
            refund, minus the service fee.
          </p>
        </div>
        <button className='mb-4 mt-2'>
          <div className='flex items-center justify-start'>
            <p
              className='text-green-850 hover:border-green-850 border-b text-green-850 border-transparent'
              style={{ fontFamily: 'airbnb-medium' }}>
              Read more about the policy
            </p>
            <div className='ml-3'>
              <svg
                style={{ fill: '#008489' }}
                viewBox='0 0 18 18'
                className='h-2 w-2'>
                <path d='m16.29 4.3a1 1 0 1 1 1.41 1.42l-8 8a1 1 0 0 1 -1.41 0l-8-8a1 1 0 1 1 1.41-1.42l7.29 7.29z' />
              </svg>
            </div>
          </div>
        </button>
      </div>
    </div>
  );
};
