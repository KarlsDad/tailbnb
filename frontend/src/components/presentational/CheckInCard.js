import React from 'react';
import Skeleton from 'react-loading-skeleton';

export default ({ loading }) => {
  return (
    <>
      <div className='w-25/31 sticky top-0 flex justify-center mt-5 ml-10'>
        <div className='w-full'>
          {loading ? (
            <div className='flex justify-start mb-4'>
              <Skeleton width={300} height={30} />
            </div>
          ) : (
            <h3
              style={{ fontFamily: 'airbnb-medium' }}
              className='w-full py-5 text-2xl text-gray-850'>
              Add dates for prices
            </h3>
          )}

          <button
            style={{ fontFamily: 'airbnb-medium' }}
            className='w-full bg-pink-850 text-white rounded-lg py-3'>
            Check Availability
          </button>
        </div>
      </div>
    </>
  );
};
