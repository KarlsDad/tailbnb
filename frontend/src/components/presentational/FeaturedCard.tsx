import * as React from 'react';
// import LazyImage from 'react-lazy-progressive-image';

interface Prop {
  img: string;
  verified: number;
  description: string;
}

export const FeaturedCard: React.FC<Prop> = ({
  img,
  verified,
  description
}) => {
  return (
    <div className='w-30/31'>
      <div className='h-full w-64 sm:w-full bg-white'>
        <img className='w-full rounded-xl' src={img} alt='adventure1' />

        <div className='flex flex-wrap items-center justify-start'>
          <p className='uppercase text-xs my-2 font-bold text-pink-700'>
            {verified}+ Verified Stays
          </p>
        </div>
        <p className='whitespace-normal my-1 text-sm tracking-wide'>
          {description}
        </p>
      </div>
    </div>
  );
};
