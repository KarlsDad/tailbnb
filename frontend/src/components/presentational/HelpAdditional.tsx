import * as React from 'react';

export const HelpAdditional: React.FC<{}> = () => {
  return (
    <div className='border-b border-gray-300'>
      <div className='flex items-center justify-between py-8'>
        <h3 style={{ fontFamily: 'airbnb-bold' }} className='text-gray-750'>
          Reservation requests
        </h3>
        <div>
          <svg
            style={{ fill: '#000000' }}
            className='w-2 h-2'
            xmlns='http://www.w3.org/2000/svg'
            viewBox='0 0 512.011 512.011'>
            <g>
              <g>
                <path
                  d='M505.755,123.592c-8.341-8.341-21.824-8.341-30.165,0L256.005,343.176L36.421,123.592c-8.341-8.341-21.824-8.341-30.165,0
			s-8.341,21.824,0,30.165l234.667,234.667c4.16,4.16,9.621,6.251,15.083,6.251c5.462,0,10.923-2.091,15.083-6.251l234.667-234.667
			C514.096,145.416,514.096,131.933,505.755,123.592z'
                />
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  );
};
