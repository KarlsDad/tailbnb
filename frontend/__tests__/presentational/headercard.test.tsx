import * as React from 'react';
import { shallow } from 'enzyme';

// Component
import { HeaderCard } from '../../src/components/presentational/HeaderCard';

describe('HeaderCard', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<HeaderCard />);
    expect(component).toMatchSnapshot();
  });
});
