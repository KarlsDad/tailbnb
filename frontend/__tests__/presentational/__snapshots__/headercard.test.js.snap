// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCard should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
